Modules/main.c

```diff
@@ -374,6 +374,7 @@ pymain_run_file_obj(PyObject *program_name, PyObject *filename,
     if (fp == NULL) {
         // Ignore the OSError
         PyErr_Clear();
+        // TODO(picnixz): strerror() is locale dependent but not PySys_FormatStderr().
```

Comment:
Then use `PyUnicode_DecodeLocale()` and add a test.

Other idea: use `str()` of the current error (before clearing it). It is not necessary `OSError`, so `errno` and `strerror()` may be wrong.

I do not think there is a simple way to test this, besides using embedded Python. So it is okay to not add a test for this.

Reply:
Do you want me to actually address this in this PR or in a follow-up PR?

Reply:
It is up to you.